R/fit.R

```diff
@@ -953,7 +998,24 @@ CmdStanMCMC <- R6::R6Class(
         out <- lapply(out, diag)
       }
       out
+    },
+    # approximate loo-cv using the loo package
+    loo = function(variables = "log_lik", r_eff = TRUE, ...) {
+      if (!requireNamespace("loo", quietly = TRUE)) {
+        stop("Please install the loo package to use this method.", call. = FALSE)
+      }
+      LLarray <- self$draws(variables)
+      if (is.logical(r_eff)) {
+        if (isTRUE(r_eff)) {
+          r_eff_cores <- list(...)[["cores"]] %||% getOption("mc.cores", 1)
+          r_eff <- loo::relative_eff(exp(LLarray), cores = r_eff_cores)
+        } else {
+          r_eff <- NULL
+        }
```

Comment:
This gives the user the choice of computing `r_eff` automatically (the default), turning off `r_eff` computation (by setting `r_eff=FALSE`) or supplying their own `r_eff` values that will be passed to loo. @avehtari This seems like the most flexible approach and if you like it I can also add the same thing in the method in RStan.

Reply:
This is good!